### PR TITLE
Improved limiting activity panels to Woo Home page

### LINKED ISSUE #857: Improve limiting activity panels to WooCommerce Home
Currently, the activity panel is properly hidden when visiting
- wp-admin pages
- wc-admin pages

and it's properly displayed when you visit directly `admin.php?page=wc-admin`

But if you first visit a wc-admin page (ie `admin.php?page=wc-admin&path=%2Fcustomers`), and then click on the menu item to go to home `admin.php?page=wc-admin` the activity panel is hidden (should be visible).

Related issue: #825

### PR DESCRIPTION
**Pull request by PanosSynetos**
closes #857

### CODE REVIEW

File: includes/class-wc-calypso-bridge-woocommerce-admin-features.php [hunk 1]
```diff
@@ -97,6 +98,53 @@ public function filter_woocommerce_body_classes( $classes ) {
...
+	 *
+	 * @return void
+	 */
+	public function filter_woocommerce_body_classes_js() {
+		?>
+		<script>
+			document.addEventListener( 'DOMContentLoaded', function() {
+
+				// Bail out earlier.
+				if ( ! document.body.classList.contains( 'woocommerce_page_wc-admin' ) ) {
```
Comment (xristos3490) [inline includes/class-wc-calypso-bridge-woocommerce-admin-features.php:114]: I believe this won't work as expected (?) 

It falls in the same pitfall as the one that engaged this PR. React will always change the page contents, not the initial body classes, as the user navigates through other react pages.

File: includes/class-wc-calypso-bridge-woocommerce-admin-features.php [hunk 2]
```diff
@@ -97,6 +98,53 @@ public function filter_woocommerce_body_classes( $classes ) {
...
+					return;
+				}
+
+				let url = location.href;
+				document.body.addEventListener( 'click', ( event ) => {
+
+					// Bail out early.
+					if ( 'A' !== event.target.tagName ) {
```
Comment (xristos3490) [inline includes/class-wc-calypso-bridge-woocommerce-admin-features.php:122]: What about `button`s? Perhaps omitting this change would work better for us. We can't be sure what triggered the page change in the DOM.

Comment (xristos3490) [reply includes/class-wc-calypso-bridge-woocommerce-admin-features.php:122]: My initial thought was to limit this script to the wp-admin context. Thus searching for the `#root` element in the DOM is enough. 

Hint: We might not even need the DOM Ready if the `#root` element exists at first paint; I would also test this perf.

**Review by xristos3490 — CHANGES_REQUESTED** [review]

File: includes/class-wc-calypso-bridge-woocommerce-admin-features.php (hunk 1 above)
Comment (PanosSynetos) [reply includes/class-wc-calypso-bridge-woocommerce-admin-features.php:114]: As we discussed

> `document.addEventListener( 'DOMContentLoaded', function() {`

Removed as it's not needed

> if ( ! document.body.classList.contains( 'woocommerce_page_wc-admin' ) ) {

Works fine, and it doesn't fall into the same pitfall.

Here's a video demonstrating that it works as expected


https://user-images.githubusercontent.com/2484390/204244914-15ffeb51-c161-4931-82e4-401de74bbee0.mp4

File: includes/class-wc-calypso-bridge-woocommerce-admin-features.php (hunk 2 above)
Comment (PanosSynetos) [reply includes/class-wc-calypso-bridge-woocommerce-admin-features.php]: Indeed, great observation about the buttons

> My initial thought was to limit this script to the wp-admin context. Thus searching for the #root element in the DOM is enough.

Checking for the existence of `woocommerce_page_wc-admin` works as expected, and there is no need for the `#root` element

**Review by xristos3490 — APPROVED** [review]
Great work, Panos! Let's merge it!

**Comment by PanosSynetos** [comment]
Cool, thanks - I'll merge it once we're ready to release it

**Pull request merged**